src/EnergyPlus/HVACUnitaryBypassVAV.cc

```diff
@@ -564,647 +556,613 @@ namespace HVACUnitaryBypassVAV {
                     ShowContinueError(state, format("Setting {} to \"None\" and the simulation continues.", cAlphaFields(19)));
-                    CBVAV(CBVAVNum).DehumidControlType = DehumidControl::None;
+                    thisCBVAV.DehumidControlType = DehumidControl::None;
                 }
             } else if (UtilityRoutines::SameString(Alphas(19), "CoolReheat")) {
-                if (CBVAV(CBVAVNum).DXCoolCoilType_Num == DataHVACGlobals::CoilDX_CoolingTwoStageWHumControl) {
-                    CBVAV(CBVAVNum).DehumidControlType = DehumidControl::CoolReheat;
+                if (thisCBVAV.DXCoolCoilType_Num == DataHVACGlobals::CoilDX_CoolingTwoStageWHumControl) {
+                    thisCBVAV.DehumidControlType = DehumidControl::CoolReheat;
                 } else {
                     ShowWarningError(state, format("Invalid {} = {}", cAlphaFields(19), Alphas(19)));
-                    ShowContinueError(state, format("In {} \"{}\".", CurrentModuleObject, CBVAV(CBVAVNum).Name));
+                    ShowContinueError(state, format("In {} \"{}\".", CurrentModuleObject, thisCBVAV.Name));
                     ShowContinueError(state, format("Valid only with {} = Coil:Cooling:DX:TwoStageWithHumidityControlMode.", cAlphaFields(14)));
                     ShowContinueError(state, format("Setting {} to \"None\" and the simulation continues.", cAlphaFields(19)));
-                    CBVAV(CBVAVNum).DehumidControlType = DehumidControl::None;
+                    thisCBVAV.DehumidControlType = DehumidControl::None;
                 }
             } else {
                 ShowSevereError(state, format("Invalid {} ={}", cAlphaFields(19), Alphas(19)));
-                ShowContinueError(state, format("In {} \"{}\".", CurrentModuleObject, CBVAV(CBVAVNum).Name));
+                ShowContinueError(state, format("In {} \"{}\".", CurrentModuleObject, thisCBVAV.Name));
             }
 
             if (NumNumbers > 8) {
-                CBVAV(CBVAVNum).minModeChangeTime = Numbers(9);
+                thisCBVAV.minModeChangeTime = Numbers(9);
             }
 
             //   Initialize last mode of compressor operation
-            CBVAV(CBVAVNum).LastMode = HeatingMode;
+            thisCBVAV.LastMode = HeatingMode;
 
-            if (CBVAV(CBVAVNum).FanType_Num != DataHVACGlobals::FanType_SimpleOnOff &&
-                CBVAV(CBVAVNum).FanType_Num != DataHVACGlobals::FanType_SimpleConstVolume &&
-                CBVAV(CBVAVNum).FanType_Num != DataHVACGlobals::FanType_SystemModelObject) {
-                ShowSevereError(state, format("{} illegal {} in fan object = {}", CurrentModuleObject, cAlphaFields(10), CBVAV(CBVAVNum).FanName));
-                ShowContinueError(state, format("Occurs in {} = {}", CurrentModuleObject, CBVAV(CBVAVNum).Name));
+            if (thisCBVAV.FanType_Num != DataHVACGlobals::FanType_SimpleOnOff &&
```

Comment:
`FanType`

Reply:
std::string FanType;                                                 // Type of fan
        int FanType_Num = 0;                                            // Fan type number (see DataHVACGlobals)